File: andre/examples/c/pass-by.c
#include <stdio.h> 

struct Person {
    int age;
    char name[10];
};

void inc_age2(struct Person *p);
void inc_age3(struct Person *p);


void increment_age(struct Person p) 
{
    p.age = p.age + 1;
}

int main()
{
    struct Person person = {40, "myname!"};

    increment_age(person);

//    printf("%d\n", person.age);
    inc_age2(&person);

    printf("%d\n", person.age);
    
    inc_age3(&person);
    
    //printf("%d\n", person.age);
    return 0;
}






void inc_age2(struct Person *p)
{
    (*p).age = (*p).age + 1;
}

void inc_age3(struct Person *p)
{
    p->age = p->age + 1;
}

File: andre/examples/go/pass-by.go
package main

import "fmt"

type Person struct {
	age  int32
	name string
}

func inc_age(p Person) {

	p.age = p.age + 1
}

func inc_age_pointer(p *Person) {

	p.age = p.age + 1
}

func main() {

	var p *Person = new(Person)

	p.age = 40
	p.name = "myname!"

	fmt.Println(p.age)

	inc_age(*p)

	fmt.Println(p.age)

	inc_age_pointer(p)

	fmt.Println(p.age)
}
